java-util/src/main/java/org/apache/druid/java/util/http/client/NettyHttpClient.java

```diff
@@ -198,11 +205,25 @@ public void messageReceived(ChannelHandlerContext ctx, MessageEvent e)
                   log.debug("[%s] Got response: %s", requestDesc, httpResponse.getStatus());
                 }
 
-                response = handler.handleResponse(httpResponse);
+                HttpResponseHandler.TrafficCop trafficCop = resumeChunkNum -> {
+                  synchronized (watermarkLock) {
+                    resumeWatermark = Math.max(resumeWatermark, resumeChunkNum);
+
+                    if (suspendWatermark >= 0 && resumeWatermark >= suspendWatermark) {
+                      suspendWatermark = -1;
+                      channel.setReadable(true);
+                      log.debug("[%s] Resumed reads from channel (chunkNum = %,d).", requestDesc, resumeChunkNum);
+                    }
+                  }
+                };
+                response = handler.handleResponse(httpResponse, trafficCop);
                 if (response.isFinished()) {
                   retVal.set((Final) response.getObj());
                 }
 
+                assert currentChunkNum == 0;
```

Comment:
did you intentionally leave it ?

Reply:
I did, it's just a thing that I think should always be true, so I put in an assert to 'document' that.